packages/utilities/src/diff/WebDiffManager.ts

```diff
@@ -0,0 +1,187 @@
+     * Singleton instance of this class
+     * @private
+     * @static
+     * @type {WebHelpManager}
+     * @memberof WebHelpManager
+     */
+    private static mInstance: WebDiffManager = null;
+
+    /**
+     * Return a singleton instance of this class
+     * @static
+     * @readonly
+     */
+    public static get instance(): WebDiffManager {
+        if (this.mInstance == null) {
+            this.mInstance = new WebDiffManager();
+        }
+
+        return this.mInstance;
+    }
+
+
+    /**
+     * Launch help page for specific group/command in browser.
+     * @param {string} inContext - content of diff to be shown
+     * @param {IHandlerResponseApi} cmdResponse - Command response object to use for output
+     * @memberof WebHelpManager
+     */
+    public async openDiffs(patchDiff: string) {
+        const doWeHaveGui = ProcessUtils.isGuiAvailable();
+        if (doWeHaveGui !== GuiResult.GUI_AVAILABLE) {
+            let errMsg = "You are running in an environment with no graphical interface." +
+                "\nAlternatively, you can run '" + ImperativeConfig.instance.findPackageBinName() +
+                " --help' for text-based help.";
+            if (doWeHaveGui === GuiResult.NO_GUI_NO_DISPLAY) {
+                errMsg += "\n\nIf you are running in an X Window environment," +
+                    "\nensure that your DISPLAY environment variable is set." +
+                    "\nFor example, type the following:" +
+                    "\n    echo $DISPLAY" +
+                    "\nIf it is not set, assign a valid value. For example:" +
+                    "\n    export DISPLAY=:0.0" +
+                    "\nThen try the --help-web option again.";
+
+                throw new ImperativeError({
+                    msg: errMsg
+                });
+            }
+
+            return;
+        }
+
+        if (this.checkWebDiffDirExists()) await new WebDiffGenerator(ImperativeConfig.instance, this.webDiffDir).buildDiffDir();
+
+        const htmlDiff = await html(patchDiff, {
+            outputFormat: "side-by-side",
+            matching: "lines",
+            diffStyle: "char",
+        });
+
+        if (htmlDiff != null) {
+            // writing the diff content into a textt file
+            fs.writeFileSync(path.join(this.webDiffDir, 'index.html'), this.genHtmlForDiffs(htmlDiff, patchDiff));
+            try {
+                ProcessUtils.openInDefaultApp(`file://${this.webDiffDir}/index.html`);
+            } catch (e) {
+                throw new ImperativeError({
+                    msg: "Failed to launch web diff, try running -h for console help instead",
+                    causeErrors: [e]
+                });
+            }
+        }
+    }
+
+
+    /**
+     * Check if the web diff direcory base has been generated
+     * at cli home
+     */
+    private async checkWebDiffDirExists() {
+        if (fs.existsSync(this.webDiffDir)) {
+            return true;
+        }
+        return false;
```

Comment:
Is there a reason for not creating the directory here when it does not exist?
Earlier code calls this function before trying to use webDiffDir. A few lines later it uses webDiffDir again, but this time it does not check for its existence.

Reply:
Yes, I think we can remove this function and check for the condition directly itself, also few lines later it doesn't check for the existence of web diff dir again because earlier it checks for it and if it doesn't exists it creates the dir.